# Training program description:

The training team at the Harvard Chan Bioinformatics Core provides bioinformatics training in multiple formats, they can be broadly divided into the following: 

1. [Introduction to Next-Generation Sequencing (NGS) analysis series](#introduction-to-next-generation-sequencing-ngs-analysis-series)
2. [Current topics in bioinformatics series](#current-topics-in-bioinformatics-series)

Our current workshops and courses are designed to help biologists become comfortable with using tools to analyse high-throughput data. We are slowly beginning to expand this repertoire to include training for researchers with more advanced bioinformatics skills. 

See our current workshop schedule on our [training website](http://bioinformatics.sph.harvard.edu/training#for-hsci-and-on-quad-hms-researchers).

## Introduction to Next-Generation Sequencing (NGS) analysis series:

### Basic Data Skills

These are short 1-1.5 day workshops that provide an introduction to computational skills required for someone to get started with analyzing high-throughput sequencing data independently. These have no prerequisites and do not require any prior experience with programming. 

| Topic |  Duration | Prerequisites 
| :----: | :----: | :----: |
| [Introduction to the command-line interface (shell)](https://hbctraining.github.io/Intro-to-Shell/) | 1 day | None |
| [Introduction to R](https://hbctraining.github.io/Intro-to-R/) | 1.5 days | None |

*Clicking on the topic name in the table above will take you to the online materials for this workshop.*

**Introduction to command-line interface (Unix/shell/Linux/bash):**

In this 1-day hands-on workshop, participants learn basic commands for navigating the file system, exploring file contents, and performing basic operations, such as moving, copying, and renaming files/folders. In addition, participants get an introduction to high-performance computing (HPC).

**Introduction to R:**

This 1.5-day workshop introduces participants to the basics of using R and RStudio, a simple programming environment that enables the effective handling of data, while providing excellent graphical support.

Participants learn how to use R for:
* exploring basic data structures
* inspecting and wrangling data
* installing & working with data analysis packages
* making publication-quality plots with the ggplot2 package

***

### Advanced Topics: Analysis of high-throughput sequencing (NGS) data

These are 2-day intensive workshops that instruct participants on how to efficiently manage and analyze data, with a focus
on the workflow for a specific type of next-generation sequencing data (i.e RNA-seq, ChIP-seq). *These workshops require participants to have taken one or more of the [Basic Data Skills workshops](#basic-data-skills) as listed in the table below.*

| Topic | Duration | Prerequisites |
| :----: | :----: | :----: |
| [Introduction to (bulk) RNA-seq using High-Performance Computing](https://hbctraining.github.io/Intro-to-rnaseq-hpc-salmon/) | 2-day | Introduction to shell |
| [Introduction to Differential Gene Expression Analysis](https://hbctraining.github.io/DGE_workshop/)  | 2-day | Introduction to R  |
| [Introduction to ChIP-seq Analysis](https://hbctraining.github.io/Intro-to-ChIPseq/) |  2-day | Introduction to R & shell |
| [Introduction to single cell RNA-seq Analysis](https://hbctraining.github.io/scRNA-seq/) | 2-day | Introduction to R |

*Clicking on the topic name in the table above will take you to the online materials for this workshop.*

**RNA-seq Analysis (from raw data to gene expression counts):**

This 2-day hands-on workshop covers the basics of bulk RNA-seq analysis; from designing a good experiment to performing QC on sequencing data to obtaining gene expression matrices. All the analysis are performed on HMS-RC's O2 cluster using temporary "training" accounts.

**Differential Gene Expression Analysis (using gene expression counts from the above workshop):**

This 2-day hands-on workshop covers the statistical considerations when using RNA-seq data for differential gene expression analysis, followed by using DESeq2 to obtain lists of differentially expressed (DE) genes. In addition, this workshop covers tools for functional analysis on the DE gene lists. All the analysis are performed using R (RStudio).

**ChIP-seq analysis:**

This 2-day hands-on workshop covers the basics of ChIP-seq analysis; from designing a good experiment to peak calling and performing a multitude of QC steps. This workshop requires participants to have a working knowledge of both R and shell, since it requires participants to use HMS-RC's O2 cluster for some of the work and R (RStudio), locally, for other analyses.

**Single cell RNA-seq:**

This workshop is in development, and we expect it to be either 2- or 3-days long. This workshop also requires participants to have a working knowledge of both R and shell. *More details coming soon!*

## [Current topics in bioinformatics series](https://hbctraining.github.io/Training-modules/):

[These short workshops](https://hbctraining.github.io/Training-modules/) (half-day or less) are designed to allow researchers, who have some familiarity with R or bash, to learn new tools and methods. 

These **free, hands-on workshops** are available to all Harvard affiliates and cover a variety of bioinformatics topics & related skills. These workshops do not require a registration, and seating in the workshops is available on a first-come-first-served basis depending on the size of the room. 

### Series theme for Fall 2019: Setting up for a successful bulk RNA-seq project

### Schedule for Fall 2019 (no pre-requisites):

| Topic | Date | Time | Location |
| :----: | :----: | :----: | :----: |
| Everything you need to know when planning an (bulk) RNA-seq analysis Part I | September 20th | 1 PM | HSPH Kresge G1 |
| Everything you need to know when planning an (bulk) RNA-seq analysis Part II | October 25th | 1 PM | HSPH Kresge G1 |
| Everything you need to know to make your data analysis reproducible | November 15th | 1 PM | HSPH Kresge G2 |
| Introduction to Version Control (Git) | December 13th | 1 PM | HSPH Kresge G2 |

* [Please click here](https://hbctraining.github.io/Training-modules/) for topics and links to materials.

# Contact us:

**Email:** [hbctraining@hsph.harvard.edu](mailto:hbctraining@hsph.harvard.edu)

**Webpage:** [http://bioinformatics.sph.harvard.edu/training/](http://bioinformatics.sph.harvard.edu/training/)

**Twitter:** [@bioinfocore](http://twitter.com/bioinfocore)
