# Training program description:

The training team at the Harvard Chan Bioinformatics Core provides bioinformatics training in multiple formats, they can be broadly divided into the following: 

1. [Introduction to High-throughput sequencing (HTS) data analysis series](#introduction-to-high-throughput-sequeuncing-hts-data-analysis-series)
2. [Current topics in bioinformatics series](#current-topics-in-bioinformatics-series)

Our current workshops and courses are designed to help biologists become comfortable with using tools to analyse high-throughput data. We are slowly beginning to expand this repertoire to include training for researchers with more advanced bioinformatics skills. 

See our current workshop schedule on our [training website](https://bioinformatics.sph.harvard.edu/upcoming-workshops).

## Introduction to high-throughput sequeuncing (HTS) data analysis series:

This series of workshops is divided into 2 categories, [Basic Data Skills](#basic-data-skills) and [Advanced Topics](#advanced-topics). The Basic workshops serve as the foundation that participants can build upon in the Advanced workshops and we will be offering these as pairs with the appropriate basic workshop preceding an advanced one. Please see below for a description of workshops under each of these two categories.

### Basic Data Skills:

These workshops provide an introduction to computational skills required for someone to get started with analyzing high-throughput sequencing data independently. These have no prerequisites and do not require any prior experience with programming.

| Topic and Link(s) to lessons | Prerequisites |
| :----: | :----: | 
| [Shell for Bioinformatics - O2 cluster](https://hbctraining.github.io/Shell-for-bioinformatics/) | None |
| [Introduction to R](https://hbctraining.github.io/Intro-to-R-flipped/schedules/links-to-lessons.html) | None |
| [Introduction to R (video tutorials)](https://projects.iq.harvard.edu/hcatrresource/) | None |
  
### Advanced Topics:

These are intensive workshops that instruct participants on how to design experiments, and efficiently manage & analyze data. They focus on the workflow for a specific type of next-generation sequencing application (i.e RNA-seq, ChIP-seq). *These workshops require participants to have taken one or more of the [Basic Data Skills workshops](#basic-data-skills) as listed in the table below.*
  
| Topic and Link(s) to lessons | Prerequisites |
| :----: | :----: |
| [Introduction to bulk RNA-seq: From reads to count matrix - O2 cluster](https://hbctraining.github.io/Intro-to-rnaseq-hpc-salmon-flipped/schedule/links-to-lessons.html) | Shell for Bioinformatics |
| [Introduction to Differential Gene Expression Analysis](https://hbctraining.github.io/Intro-to-DGE/)  | Introduction to R  |
| [Investigating chromatin biology using ChIP-seq and CUT&RUN - O2 cluster](https://hbctraining.github.io/Investigating-chromatin-biology-ChIPseq/) | Shell for Bioinformatics |
| [Introduction to single cell RNA-seq](https://hbctraining.github.io/scRNA-seq_online/schedule/links-to-lessons.html) | Introduction to R |
| [Introduction to Variant Analysis](https://hbctraining.github.io/Intro-to-variant-analysis/) | Shell for Bioinformatics |
| [Tools for Reproducible Research](https://hbctraining.github.io/Tools-for-reproducible-research/) | Introduction to R |
| [Pseudobulk and related approaches for scRNA-seq analysis](https://hbctraining.github.io/Pseudobulk-for-scRNAseq/) | Introduction to R |
| [Introduction to Peak Analysis]([https://hbctraining.github.io/Tools-for-reproducible-research/](https://github.com/hbctraining/Intro-to-peak-analysis)) | Introduction to R |

***

## Current topics in bioinformatics series:

These workshops provide instruction on basic data skills as well as introduce new topics of interest to the community.

### R-based short workshops:

| Topic and Link(s) to lessons | Prerequisites |
|:---------------|:-------------:|
| [Introduction to R](https://hbctraining.github.io/Training-modules/IntroR) | None |
| [Introduction to R Practical](https://hbctraining.github.io/Training-modules/IntroR_practical_online_resource) | Beginner R or [Completion of the Intro to R online resource](https://projects.iq.harvard.edu/hcatrresource/) |
| [Functional analysis of gene lists](https://hbctraining.github.io/Training-modules/DGE-functional-analysis) | Beginner R or [Intro to R workshop](https://hbctraining.github.io/Training-modules/IntroR) |
| [Reproducible Research using RMarkdown](https://hbctraining.github.io/Training-modules/Rmarkdown) | Beginner R or [Intro to R workshop](https://hbctraining.github.io/Training-modules/IntroR) |
| [Publication Perfect I: Data visualization basics with ggplot2](https://hbctraining.github.io/Training-modules/publication_perfect) | Beginner R or [Completion of the Intro to R online resource](https://projects.iq.harvard.edu/hcatrresource/) |
| [Publication Perfect II: Figure formatting in R](https://hbctraining.github.io/Training-modules/publication_perfect#part-ii) | [Publication Perfect: Part I](https://hbctraining.github.io/Training-modules/publication_perfect) |
| [Interact with your data using RShiny](https://hbctraining.github.io/Training-modules/RShiny/) | Beginner R or [Completion of the Intro to R online resource](https://projects.iq.harvard.edu/hcatrresource/) |


### Shell-based short workshops:

| Topic and Link(s) to lessons | Prerequisites |
|:---------------|:-------------:|
| [Basic Shell/Introduction to the command-line interface](https://hbctraining.github.io/Training-modules/Basic_shell) | None |
| [Intermediate Shell/Accelerate with Automation](https://hbctraining.github.io/Training-modules/Accelerate_with_automation/) | [Basic Shell](https://hbctraining.github.io/Training-modules/Basic_shell) |
| [Advanced Shell/Finding and Summarizing Data from Colossal Files](https://hbctraining.github.io/Training-modules/Finding_and_summarizing_colossal_files/) | [Basic Shell](https://hbctraining.github.io/Training-modules/Basic_shell) |
| [Tips and Tricks on O2](https://hbctraining.github.io/Training-modules/Tips_and_Tricks_on_O2/) | [Basic Shell](https://hbctraining.github.io/Training-modules/Basic_shell) |
| [“Track Changes” for Your Code: An Introduction to Git and GitHub]() | No pre-requisite (GitHub Desktop) |
| [Coding with Others: Managing Conflicts on GitHub]() | “Track Changes” for Your Code |
| [Accessing genomic reference and experimental sequencing data](https://hbctraining.github.io/Accessing_public_genomic_data) | [Basic Shell](https://hbctraining.github.io/Training-modules/Basic_shell)  |


### Other short workshops:

| Topic and Link(s) to lessons | Prerequisites |
|:---------------|:-------------:|
| [Introduction to Python](https://hbctraining.github.io/Training-modules/Python) | None |
| [Planning a bulk RNA-seq analysis: Part I](https://hbctraining.github.io/Training-modules/planning_successful_rnaseq#part-i) | None |
| [Planning a bulk RNA-seq analysis: Part II](https://hbctraining.github.io/Training-modules/planning_successful_rnaseq#part-ii) | None |
| [Make your (RNA-seq) data analysis reproducible](https://hbctraining.github.io/Training-modules/reproducible_analyses)- *Taught by [Julie Goldman](https://scholar.harvard.edu/julie_goldman) from Countway Library* | None |
| [Improving your (RNA-seq) data analysis using version control (Git) - In collaboration with HBC-RCS](https://hbctraining.github.io/versioning_data_scripts/) | None |

***

# Contact us:

**Email:** [hbctraining@hsph.harvard.edu](mailto:hbctraining@hsph.harvard.edu)

**Webpage:** [http://bioinformatics.sph.harvard.edu/training/](http://bioinformatics.sph.harvard.edu/training/)

