The training team at the Harvard Chan Bioinformatics Core provides training to help biologists become comfortable with using bioinformatics tools to analyse high-throughput sequencing (HTS) data. 

We offer courses and skills at three different levels starting at the basics and building upwards. We focus on the two most commonly used HTS interfaces, **R** and **Bash/Shell**.

<p align="center">
<img src="img/Teaching_organization_greyscale_exterior_arrows_questions.png" width="800">
</p>

## What are the basic skills I need?

<table>
  <tr>
    <th>Skill</th>
    <th>Who needs it</th>
    <th>Overview</th>
    <th>Courses</th>
  </tr>
  <tr>
    <td><b>A1</b> - Using the command line interface</td>
    <td>Anyone planning on doing scientific computing using the command-line.</td>
    <td>Understanding the need for shell and master basic commands</td>
    <td><ul><li><a href="https://hbctraining.github.io/Shell-for-bioinformatics/">Shell for Bioinformatics</a></li>
     </ul></td>
  </tr>
  <tr>
    <td><b>A2</b> - Using a HPC cluster</td>
    <td>Anyone who wants to efficiently run analyses on large datasets (requiring more computational resources than a laptop can provide).
</td>
    <td>Understanding the components of a high performance compute cluster (HPC), and learning to navigate and properly use available HPCs at Harvard.</td>
    <td><ul><li><a href="https://hbctraining.github.io/Shell-for-bioinformatics/">Shell for Bioinformatics</a></li>
      </ul></td>
  </tr>
  <tr>
    <td><b>B</b> - Using R </td>
    <td>Anyone who wants to learn a programming language that is especially useful for data wrangling and statistics.</td>
    <td>Learning the R and RStudio interface, Basic R syntax, and data visualization</td>
    <td><ul><li><a href="https://hbctraining.github.io/Intro-to-R-flipped/">Introduction to R</a></li>
      <li><a href="https://projects.iq.harvard.edu/hcatrresource/">Introduction to R (video tutorials)</a></li></ul></td>
  </tr>
</table>

***

## How can I apply the basic skills?

<table>
  <tr>
    <th>Skill</th>
    <th>Who needs it</th>
    <th>Overview</th>
    <th>Courses</th>
  </tr>
  <tr>
    <td><b>C</b> - Analysis of HTS data in the HPC environment</td>
    <td>Anyone planning on doing genomic or transcriptomic next-generation sequencing and is interested in analyzing their own data.</td>
    <td> - Analysis of bulk RNAseq data, Variant Analysis, and sequencing data related to Chromatin biology starting with raw data.<br><br>
      - Automating the workflow with advanced shell scripts.</td>
    <td><ul><li><a href="https://hbctraining.github.io/Intro-to-bulk-RNAseq/">Introduction to bulk RNA-seq: From reads to count matrix</a></li> 
      <li><a href="https://hbctraining.github.io/Investigating-chromatin-biology-ChIPseq/">Investigating chromatin biology using ChIP-seq and CUT&RUN</a></li>
      <li><a href="https://hbctraining.github.io/Intro-to-variant-analysis/">Introduction to Variant Analysis</a></li></ul></td>
  </tr>
  <tr>
    <td><b>D</b> - Statistical Analysis of HTS data in R</td>
    <td>Anyone who wants to use popular R packages for downstream analysis of HTS data. Main focuses include Seurat and DESeq2.</td>
    <td> - Using R to implement best practices workflows for the analysis of various forms of HTS data.<br><br>
      - Clear explanations of the theory behind each step in of the workflow.</td>
    <td><ul><li><a href="https://hbctraining.github.io/Intro-to-scRNAseq/">Introduction to single cell RNA-seq Analysis</a></li>
      <li><a href="https://hbctraining.github.io/Intro-to-DGE/">Introduction to Differential Gene Expression Analysis</a></li>
      <li><a href="https://hbctraining.github.io/Intro-to-peak-analysis/">Introduction to Peak Analysis</a></li>
      <li><a href="https://hbctraining.github.io/Pseudobulk-for-scRNAseq/">Pseudobulk and related approaches for scRNA-seq analysis</a></li></ul></td>
  </tr>
</table>

***  
  
## How can I build my skillset further?

<table>
  <tr>
    <th>Skill</th>
    <th>Who needs it</th>
    <th>Overview</th>
    <th>Courses</th>
  </tr>
  <tr>
    <td><b>E</b> - Advanced programming with the <code>bash</code> command line</td>
    <td>Anyone who wants to create custom shell scripts and utilize bash for various tasks.</td>
    <td>Learning to include version control in your projects and advanced bash scripting</td>
    <td><ul><li><a href="https://hbctraining.github.io/versioning_data_scripts/">Improving your (RNA-seq) data analysis using version control (Git)</a></li>
      <li><a href="https://hbctraining.github.io/Accessing_public_genomic_data/">Accessing genomic reference and experimental sequencing data</a></li></ul></td>
  </tr>
  <tr>
    <td><b>F</b> - Advanced R for generating complex plots and reports</td>
    <td>Anyone who wants to make publication quality figures<br><br>
      Anyone who wants to make high level HTML reports of analyses</td>
    <td>Exploring additional R features such as reports and publication perfect figures</td>
    <td><ul><li><a href="https://hbctraining.github.io/publication_perfect/">Publication Perfect</a></li>
      <li><a href="https://hbctraining.github.io/Training-modules/Rmarkdown">Generating research analysis reports with RMarkdown</a></li></ul></td>
  </tr>
</table>   

***  
  
  
## Additional Courses
- [Introduction to Python](https://hbctraining.github.io/Training-modules/Python)  
  
  ***

# Contact us:

**Email:** [hbctraining@hsph.harvard.edu](mailto:hbctraining@hsph.harvard.edu)

**Webpage:** [http://bioinformatics.sph.harvard.edu/training/](http://bioinformatics.sph.harvard.edu/training/)

*These materials have been developed by members of the teaching team at the [Harvard Chan Bioinformatics Core (HBC)](http://bioinformatics.sph.harvard.edu/) RRID:SCR_025373. These are open access materials distributed under the terms of the [Creative Commons Attribution license](https://creativecommons.org/licenses/by/4.0/) (CC BY 4.0), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.*

*A lot of time and effort went into the preparation of these materials. Citations help us understand the needs of the community, gain recognition for our work, and attract further funding to support our teaching activities. Thank you for citing the corresponding course (as suggested in its "Read Me" section) if it helped you in your data analysis.*
  
<!-- This content will not appear in the rendered Markdown -->
<!-- Removed links to FAS-RC:
 <li><a href="https://hbctraining.github.io/Intro-to-shell-fasrc-flipped/">Introduction to the command-line interface (shell) - FAS-RC cluster</a></li>
 <li><a href="https://hbctraining.github.io/Intro-to-rnaseq-fasrc-salmon-flipped/">Introduction to (bulk) RNA-seq using High-Performance Computing - FAS-RC cluster</a></li> -->
