## Bioinformatics Trainer and Analyst Position

|             |    |
|:------------------------:|:-----------------------------------------:|
| **Title:**  | Research Associate - Bioinformatics Trainer and Analyst| 
| **School:**  |  [Harvard T.H. Chan School of Public Health](https://www.hsph.harvard.edu/), Boston MA| 
| **Department/Area:** |   Biostatistics| 


> ***This will be a remote (telework) position until the end of the COVID-19 pandemic.***

## About us
The [Harvard Chan Bioinformatics Core (HBC)](https://bioinformatics.sph.harvard.edu/) is a small team of skilled and personable bioinformaticians. We are a central resource for bioinformatics research, services and training at Harvard and across the Boston biomedical community. We work closely with biomedical scientists to develop and execute innovative workflows to analyze, interpret, visualize and distribute scientific discoveries derived from the analysis of high-throughput data.

As a team, we embrace an open source approach to computing and training. We use and teach open source tools and contribute to the development of open source projects; our training materials are freely available on [GitHub (https://hbctraining.github.io/main)](https://hbctraining.github.io/main). In line with this community approach, the HBC emphasizes teamwork, collaboration and provides a supportive environment where team members can learn from each other.


## Who are we looking for?

The Harvard Chan Bioinformatics Core is excited to **expand our bioinformatics training program** as part of a new collaboration with the [Dana-Farber / Harvard Cancer Center (DF/HCC)](https://www.dfhcc.harvard.edu/). This role provides a unique and rewarding opportunity to train and support world-class researchers making a profound impact on human health. We are looking for a bioinformatician to join our team in our efforts to provide education and analytical support to the Harvard community. The ideal candidate is:

* Someone who is enthusiastic about teaching (as demonstrated by their teaching experience)
* Someone who enjoys working in a collaborative environment
* Someone who has a background in high-throughput data analysis, specifically for next-generation sequencing (NGS) data


## What will you be doing?

The majority of your time will be dedicated to your role **as a member of the training team**, led by the HBC Training Director. In this role, you will:

* Help teach workshops and courses (i.e. basic data skills, NGS analysis) geared towards graduate students, postdocs, research staff and faculty from the greater Harvard community 
* Generate new content and update existing content based on the latest best practices
* Assist with organizational tasks related to training activities
* Be fully involved in determining the future direction of the training program

As a member of the HBC you will also be **providing analysis support to researchers** at Harvard and in the broader Boston biomedical community under the direction of the Coreâ€™s Director and Associate Director. In this role, you will:

* Support selected research projects by analyzing high-throughput data with best practice approaches used in the Core, assessing new methods where appropriate
* Document all work thoroughly, and provide clear, manuscript-level reports of analyses and results
* Be responsible for good data management of your projects
* Participate in Core meetings, liaising with collaborating researchers, and participate in developing manuscripts for publication (when the opportunity presents)


### Basic Qualifications	
* **Doctoral degree** in one of the following biomedical research areas:
  * Molecular/cellular biology
  * Cancer biology
  * Biostatistics
  * Computational biology
  * Bioinformatics
* At least **5 years of bioinformatics experience (including doctoral work)** and at least 2 years since completion of your doctoral degree
* **Proven experience with teaching courses/workshops with hands-on coding components**
* Excellent written and spoken English
* Programming and data skills (Python, R, shell, version control)
* Working knowledge of biology and genetics
* Experience in **at least one** of the following NGS domains:
  * Whole genome sequencing
  * Exome-seq
  * RNA-seq (bulk/single-cell)
  * ChIP-seq or ATAC-seq
  * Other sequencing applications

### Additional Qualifications	
* Proven ability to interpret and analyze large data sets and present results
* Strong interpersonal skills
* Basic knowledge of statistics
* Knowledge of and experience with cancer datasets is a plus
	

## How do I apply?

To apply **[Click Here](https://academicpositions.harvard.edu/postings/9711)**!


## Questions?

### Contact us!
Email	**[hbctraining (at) hsph.harvard.edu](mailto:hbctraining@hsph.harvard.edu)** for more information


> ### Equal Opportunity Employer	
> We are an equal opportunity employer and all qualified applicants will receive consideration for employment without regard to race, color, religion, sex, national origin, disability status, protected veteran status, gender identity, sexual orientation, pregnancy and pregnancy-related conditions or any other characteristic protected by law.

--- 


### What I removed:
	
> NOTE: I took this out because they can use the link provided above for more detail on the courses

Training topics include basic data skills (shell, version control, R, high-performance computing, Python, etc.), data management, and NGS data analysis (RNA-seq, ChIP-seq, variant calling, etc.). In addition to teaching existing content at workshops, you will 


> NOTE: consider adding some of these back in as bullet points under "Who we are looking for?" Imo, having all of these is overkill

You have a background in cancer biology, biomedical or quantitative science and a strong interest in helping biomedical researchers. You have experience with NGS data analysis and enjoy teaching. You thrive on scientific challenges, love sharing knowledge, and enjoy working both collaboratively and independently. You excel at communicating with programmers and wet-lab scientists alike, and are able to explain complex concepts simply in written and spoken form. You are motivated to continually expand your skills and are keen to learn and apply new methods. You have a system for writing good code and managing your data, and see value in enabling reproducible research. You are organized, have strong time management skills and are capable of simultaneously working on multiple projects and meeting deadlines.

> NOTE: I don't think any of this text is necessary since it is in the main offical post

Our training team is tasked with the mission of educating the Harvard community on best practices in experimental design, analysis workflows, and data management. Workshops are designed to foster independence and confidence among participants, many of whom are wet-lab biologists applying computational methods to their research for the first time.

The HBC also provides consulting services for researchers wishing to analyze high-throughput biomedical datasets. We work closely with investigators across a broad range of disciplines from the Harvard and with industry collaborators. Projects range from short-term projects (weeks to months) to longer term collaborations (months to years) involving large data sets or multiple omics data types. By combining training and consulting activities, our training team members continue to develop their expertise in NGS data analysis through access to real-world data sets and opportunities to apply emerging and best practice methods.



