# Pseudobulk and related approaches for scRNA-seq analysis

<!-- OLD: # Differential expression of single cell RNA-seq workshop -->

*Note: Some HBC workshops are taught in person and others are online via Zoom. Please take note for scheduling purposes!*
<!-- *Note: In 2024, some workshops will be taught in person and others will be online via Zoom.* -->


### ** Workshop dates and times:**
**IN-PERSON (Longwood Medical Area):** 

**Tuesday, October 21, 2025**: 9:30 AM - 12 PM

**Friday, October 24, 2025**: 9:30 AM - 12 PM

**Tuesday, October 28, 2025**: 9:30 AM - 12 PM

**Friday, October 31, 2025**: 9:30 AM - 12 PM
-->

NOTE: Workshop dates are subject to change - please check back

_Instruction will be mostly learner-centric, requiring a time commitment between the workshop sessions!_

---

### **Description:**
After clustering cells as part of a single-cell RNA-seq experiment, investigators are often interested in carrying out a differential expression analysis between conditions within certain cell types. When the experimental design includes replicates with multi-conditions there is power in identifying differentially expressed genes using a pseudbulk approach.  In this hands-on workshop, we will:

- Discuss various approaches for differential expression using single cell RNA-seq data
- Prepare single-cell RNA-seq raw count data for pseudobulk analysis
- Perform differential expression analysis on pseudobulk counts using DESeq2
- Present approaches for evaluating differential proportions of cells between conditions


---

### **Prerequisites:**

This is one of our advanced [workshops](https://hbctraining.github.io/main/), and **requires registrants to have attended the following Basic workshop** offered by our training team in the **last 6-8 months**:  

- [Introduction to R](https://hbctraining.github.io/main/registrations/AllFunders_Intro-to-R)

There is an expectation of participants to have some/all of the following programming knowledge:

* Solid understanding of **data structures** and fluent implementation of **basic syntax** in R.
* Ability to use **functions** regularly and can modify arguments when necessary.
* Comfortable **installing packages and troubleshooting** issues.
* **Data wrangling** expertise (subsetting vectors and data frames) using base R or tidyverse
* Familiarity with ggplot2 for **data visualization**.
* Familiarity with RStudio interface

_Participants in this workshop would benefit from having some familiarity with single-cell RNA-seq analysis and the use of the [R Seurat package](https://satijalab.org/seurat/), but it is not a requirement._

**I am fluent in R but have not attended the HBC prerequisite workshop, can I still register?**

Yes, please register and provide your R/programming exerience on the form, and we'll get back to you.

<!--
**AND** please do the following:

- Complete the registration first
- Then email us directly at hbctraining@hsph.harvard.edu with a description of your experience/usage of R. -->

---

### **Who should attend?**

[Eligible*](#eligibility-requirements) Harvard researchers who want to learn: 

- How to organize scRNA-seq data for DE analysis
- How to use DESeq2 to obtain a list of differentially expressed genes
- How to visualize expression patterns of differentially expressed genes in single cell RNA-seq data


---

### **Cost:**

There is a ***non-refundable*** and ***non-transferable*** $65 registration fee for this advanced workshop.

We will be accepting 25 participants on a first-come, first-served basis:

- **If you are one of the first 25 eligible* registrants**, you will receive an email within 1 week with a link to pay the (non-refundable & non-transferable) $65 registration fee. 
- **If you are not among the first 25 eligible* registrants**, you will be added to the waitlist and notified when we open registration for the next iteration of this workshop.

*NOTE: You will not have a reserved seat for this workshop until you pay the registration fee. Please make sure you pay within the time stated in that email, else you will lose your spot to someone on the waitlist.*

*NOTE: HMS Graduate Students may receive waivers on registration fees. Please register and weâ€™ll followup with you.

---

### **Registration:**

Please make sure you **take the following into account when signing up** for the workshop:

1. _Mandatory attendance for all four classes._
2. _2-3 hours of reading and exercises from selected lessons before each workshop sessions._
3. _Active participation during discussion sessions._

<div style="text-align:center">
	 <a><button name="button" style = "color: red" >Registration is now closed!</button></a>
</div>

<div style="text-align:center"> 
	
(Registration opens 2 - 3 weeks before the start of advanced workshops.)
</div> 

<!--
<div style="text-align:center">
	 <a><button name="button" style = "color: blue" onclick="location.href='https://harvard.az1.qualtrics.com/jfe/form/SV_88Qi9mDKwHqz3hA'">Click here to Register!</button></a>
</div>

<div style="text-align:center">
	 (Please check the eligibility requirements below prior to registering)
</div> 
-->

<!-- This content will not appear in the rendered Markdown -->


---

### **Eligibility requirements:**

<sup>*</sup>To be eligible to attend this workshop you should fulfill at least one of the following criteria:

- Harvard Medical School researcher in a lab on the Quad, with grants administered by HMS Gordon Hall
- Harvard School of Dental Medicine researcher, with grants administered by HMS Gordon Hall
- Graduate student at the Harvard Medical School
- Researcher affiliated with the [Dana-Farber / Harvard Cancer Center (DF/HCC](https://www.dfhcc.harvard.edu)).
- Researcher at a Harvard-affiliated institution participating in HIV related research [(CFAR)](https://cfar.globalhealth.harvard.edu/).

*If you are unsure of your eligibility, please register anyway and we will get back to you.*


NOTE: We do ***not*** record our training sessions. 

---

### **Questions?**

Please email us at hbctraining@hsph.harvard.edu with any questions.
