# Pseudobulk and related approaches for scRNA-seq analysis

*Note: Some HBC workshops are taught in person and others are online via Zoom. Please take note for scheduling purposes!*

<!-- ### **IN-PERSON (Longwood Medical Area) Workshop dates and times:**
### **ONLINE (Zoom) Workshop dates and times:** -->

**TBD**: 10 AM - 12 PM

**TBD**: 10 AM - 12 PM

**TBD**: 10 AM - 12 PM

**TBD**: 10 AM - 12 PM


NOTE: Workshop dates are subject to change - please check back

_Instruction will be mostly learner-centric, requiring a time commitment between the workshop sessions!_

---

### **Description:**
After clustering cells as part of a single-cell RNA-seq experiment, investigators are often interested in carrying out a differential expression analysis between conditions within certain cell types. When the experimental design includes replicates with multi-conditions there is power in identifying differentially expressed genes using a pseudbulk approach.  In this hands-on workshop, we will:

- Discuss various approaches for differential expression using single cell RNA-seq data
- Prepare single-cell RNA-seq raw count data for pseudobulk analysis
- Perform differential expression analysis on pseudobulk counts using DESeq2
- Present approaches for evaluating differential proportions of cells between conditions

---

### **Prerequisites:**

This is one of our advanced [workshops](https://hbctraining.github.io/main/), and **requires registrants to have attended the following Basic workshop** offered by our training team in the **last 6-8 months**:  

- [Introduction to R](https://hbctraining.github.io/main/registrations/AllFunders_Intro-to-R)

There is an expectation of participants to have some/all of the following programming knowledge:

- Solid understanding of **data structures** and fluent implementation of **basic syntax** in R.
- Ability to use **functions** regularly and can modify arguments when necessary.
- Comfortable **installing packages and troubleshooting** issues.
- **Data wrangling** expertise (subsetting vectors and data frames) using base R or tidyverse
- Familiarity with ggplot2 for **data visualization**.
- Familiarity with RStudio interface

_Participants in this workshop would benefit from having some familiarity with single-cell RNA-seq analysis and the use of the [R Seurat package](https://satijalab.org/seurat/), but it is not a requirement._

**I am fluent in R but have not attended the HBC prerequisite workshop, can I still register?**

Yes, please register and provide your R/programming exerience on the form, and we'll get back to you.

---

### **Who should attend?**

Researchers who want to learn: 

- How to organize scRNA-seq data for DE analysis
- How to use DESeq2 to obtain a list of differentially expressed genes
- How to visualize expression patterns of differentially expressed genes in single cell RNA-seq data

---

### **Cost:**

There is a ***non-refundable*** and ***non-transferable*** registration fee for this advanced workshop*. After registering below, you will receive an email within 1 week with a link to pay.

Priority is given to Harvard Quad and DF/HCC researchers, who are subsidized for this workshop at a price of $65. Others interested in our workshops can find [our prices here](https://hsph.harvard.edu/research/bioinformatics/training/workshop-pricing/) and should register below.

> **NOTE**: _You will not have a reserved seat for this workshop until you pay the registration fee_. Please make sure you pay within the time stated in that email, else you will lose your spot to someone on the waitlist.

> **NOTE*: HMS Graduate Students may receive waivers on registration fees. Please register and weâ€™ll followup with you.

---

### **Registration:**

Please make sure you **take the following into account when signing up** for the workshop:

1. _Mandatory attendance for all four classes._
2. _3-5 hours of reading and exercises from selected lessons before each workshop sessions._
3. _Active participation during discussion sessions._

<div style="text-align:center">
	 <a><button name="button" style = "color: red" >Registration is now closed!</button></a>
</div>

<div style="text-align:center"> 
	
(Registration opens 2 - 3 weeks before the start of advanced workshops.)
</div> 

<!-- This content will not appear in the rendered Markdown -->
<!--

<div style="text-align:center">
	 <a><button name="button" style = "color: blue" onclick="location.href='https://harvard.az1.qualtrics.com/jfe/form/SV_88Qi9mDKwHqz3hA'">Click here to Register!</button></a>
</div>

<div style="text-align:center">
	 (Please check the eligibility requirements below prior to registering)
</div> 
-->

---

### **Eligibility:**

Priority is given to Harvard-affiliated researchers:
- at the Harvard Medical School in a lab on the Quad, with grants administered by HMS OR
- at the Harvard School of Dental Medicine, with grants administered by HMS Gordon Hall OR
- an HMS graduate student OR 
- affiliated with [Dana-Farber / Harvard Cancer Center (DF/HCC](https://www.dfhcc.harvard.edu)) OR
- an AIDS researcher affiliated with Harvard Center for AIDS Research (CFAR)

*If you are unsure of your eligibility, please register anyway and we will get back to you.*


NOTE: We do ***not*** record our training sessions. 

---

### **Questions?**

Please email us at hbctraining@hsph.harvard.edu with any questions.
